dali/imgcodec/util/convert_gpu.cu

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <vector>
+#include "dali/imgcodec/util/convert_gpu.h"
+#include "dali/imgcodec/util/convert.h"
+#include "dali/kernels/slice/slice_flip_normalize_permute_pad_gpu.h"
+#include "dali/kernels/dynamic_scratchpad.h"
+#include "dali/kernels/imgproc/color_manipulation/color_space_conversion_kernel.cuh"
+
+namespace dali {
+
+template<> float16 max_value<float16>() {
+  assert(false);
```

Comment:
Why? Could you elaborate in && "...", please?

Reply:
Sure. It is never called, but there had to be some overload provided because there is no c++17 in cuda files (missing the if constexpr), line 42 in the same file.